src/components/withResource.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
```

Comment:
Please merge this PR into master. You are going to run into all sorts of problems as master advances farther and farther away from your feature branch. You're also asking your reviewers to review code which has already been reviewed and merged (i.e., this file).